lib/Graph.js

```diff
@@ -216,6 +219,7 @@ Graph.prototype.literal = function (val) {
   if (typeof val == 'undefined') return LITERAL_UNDEFINED
   if (val == null) return LITERAL_NULL
 
+  // This isn't guaranteed to be unique. We should consider switching to unique keys
```

Comment:
Why not use oidProvider here too?

Reply:
This is where I think we handle graph nodes that are literals like `true`, numbers, and strings. Of those, maybe strings could be keys in WeakMaps, but we would want to map them based on their characters rather than if they are the same object instance. I think. I also just didn't want to touch too much.

Reply:
Actually, looking at the code a bit more, this is also called for object literals, so those would fall victim to the v8 problem. I'm going to see it I can special case for that.

Reply:
i like the idea of just special-casing undefined, null, boolean, number and doing the rest with WeakMap. does that mean you can remove oid???

Reply:
PTAL at the new commit, which also addresses the object ids of object literals. We still use `oid` to get hashes for numbers and strings. However, there is still a (much smaller now) probability of collision. The `oid` hash of a string could collide with the incremental object id of an object literal in this `graph._literalFns` map. I left a comment about what I think could alleviate any collisions -- a separate map for just numbers and strings, but also not sure I want to mess with that rn, unless you were thinking of something else?

Reply:
oh i didn't know that strings can't be keys in weakmaps...this is probably...ok. i expect you don't have many strings in your shepherd graph anyway.

Reply:
a separate map seems like a fine strategy too